rasterio/zarr.py

```diff
@@ -0,0 +1,72 @@
+"""Zarr storage"""
+
+from collections.abc import MutableMapping
+import json
+import logging
+from pathlib import Path
+
+import numpy
+from rasterio.windows import Window
+
+log = logging.getLogger(__name__)
+
+
+class RasterioStore(MutableMapping):
+    def __init__(self, dataset):
+        self.dataset = dataset
+        chunk_height, chunk_width = self.dataset.block_shapes[0]
+        self._data = {
+            ".zgroup": json.dumps({"zarr_format": 2}).encode("utf-8"),
+            Path(self.dataset.name).name
+            + "/.zarray": json.dumps(
+                {
+                    "zarr_format": 2,
+                    "shape": (
+                        self.dataset.count,
+                        self.dataset.height,
+                        self.dataset.width,
+                    ),
+                    "chunks": (
+                        1,
```

Comment:
For this first revision, I chose to put each band in its own chunk. Not ideal for all situations, I'm sure. I've got a lot to learn about zarr in production.

Reply:
For uncompressed flat binary data, you can break the bytes into chunks however you want. Perhaps this could be a user configurable parameter.